common/help-on-spot-models/src/utils/Database.ts

```diff
@@ -1,41 +1,57 @@
-import { Connection, createConnection } from 'typeorm';
-import {PostgresConnectionOptions} from "typeorm/driver/postgres/PostgresConnectionOptions";
+import { Connection, ConnectionManager, ConnectionOptions, createConnection, getConnectionManager } from 'typeorm'
+import {ormConfig} from "../ormConfig";
 import User from "../entity/User";
-import Address from "../entity/Address";
 import Qualification from "../entity/Qualification";
 import Organisation from "../entity/Organisation";
+import Address from "../entity/Address";
 import Request from "../entity/Request";
 import RequestResponse from "../entity/RequestResponse";
-import {ormConfig} from "../ormConfig";
 import {QualificationsMigration1586981133000} from "../migration/qualificationsMigration";
 
 export class Database {
-  private connection?: Connection;
+  private connectionManager: ConnectionManager
+
+  constructor() {
+    this.connectionManager = getConnectionManager()
+  }
+
+  public async getConnection(): Promise<Connection> {
+    const CONNECTION_NAME = `default`
+
+    let connection: Connection
+
+    if (this.connectionManager.has(CONNECTION_NAME)) {
+      console.info(`Database.getConnection()-using existing connection ...`)
+      connection = await this.connectionManager.get(CONNECTION_NAME)
+
+      if (!connection.isConnected) {
+        connection = await connection.connect()
+      }
+    }
+    else {
+      console.log(`Database.getConnection()-creating connection ...`)
 
-  public async connect() {
-    try {
-      this.connection = await createConnection({
+      const connectionOptions: ConnectionOptions = {
         ...ormConfig,
+        name: `default`,
+        type: `postgres`,
+        synchronize: true,
         entities: [User, Qualification, Organisation, Address, Request, RequestResponse],
         migrations: [ QualificationsMigration1586981133000 ],
-        schema: 'public',
-        type: 'postgres',
-      } as PostgresConnectionOptions);
-    } catch (e) {
-      console.log(e);
+      }
+      connection = await createConnection(connectionOptions)
+      const migrations = await connection.runMigrations();
+      migrations.forEach((migration) => {
+        console.log('Executed migration', migration);
+      });
     }
-    console.info('Running migrations');
-    const migrations = await this.connection!.runMigrations();
-    migrations.forEach((migration) => {
-      console.info('Executed migration', migration);
-    });
 
-    return this.connection;
+    return connection
   }
 
-  public async disconnect() {
-    if (this.connection) {
-      await this.connection.close();
+  public async disconnect(connection: Connection) {
+    if (connection) {
```

Comment:
Would probably more safe to remove the param and just close every connection found in the manager?

Reply:
I don't know if this potentially could interfere with other running lambdas or lambda instances?